tests/gui/test_logic.py

```diff
@@ -0,0 +1,100 @@
+import platform
+import subprocess
+from unittest import mock
+
+import pytest
+
+from dangerzone.gui.logic import DangerzoneGui
+
+if platform.system() == "Linux":
+    from xdg.DesktopEntry import DesktopEntry
+
+
+@pytest.mark.skipif(platform.system() != "Linux", reason="Linux-only test")
+def test_order_mime_handers() -> None:
+    """
+    Given a default mime handler returned by xdg-mime,
+    ensure it is the first item available in the list
+    of compatible applications.
+    """
+    mock_app = mock.MagicMock()
+    dummy = mock.MagicMock()
+
+    mock_desktop = mock.MagicMock(spec=DesktopEntry)
+    mock_desktop.getMimeTypes.return_value = "application/pdf"
+    mock_desktop.getExec.side_effect = [
+        "/usr/bin/madeup-evince",
+        "/usr/local/bin/madeup-mupdf",
+        "/usr/local/bin/madeup-libredraw",
+    ]
+    mock_desktop.getName.side_effect = [
```

Comment:
:) See above - the explanation is actually applicable for getName moreso than for getExec, it doesn't really matter what the exec value is, but I used side_effect consistently so that the test output was more readable